test/test_rospy/nodes/listener.py

```diff
@@ -55,7 +55,7 @@ def listener():
     # run simultaenously.
     rospy.init_node('listener', anonymous=True)
 
-    rospy.Subscriber("chatter", String, callback)
+    sub = rospy.Subscriber("chatter", String, callback)
```

Comment:
was the subscriber getting garbage collected before?

Reply:
Yes, that is my assumption.